test/framework/easyconfig.py

```diff
@@ -1132,6 +1133,51 @@ def test_hide_hidden_deps(self):
         self.assertEqual(ec['hiddendependencies'][0]['full_mod_name'], 'toy/.0.0-deps')
         self.assertEqual(ec['dependencies'], [])
 
+    def test_quote_str(self):
+        """Test quote_str function."""
+        teststrings = {
+            'foo' : '"foo"',
+            'foo\'bar' : '"foo\'bar"',
+            'foo\'bar"baz' : '"""foo\'bar"baz"""',
+            "foo'bar\"baz" : '"""foo\'bar"baz"""',
+            "foo\nbar" : '"foo\nbar"'
+        }
+
+        for t in teststrings:
+            self.assertEqual(quote_str(t), teststrings[t])
+
+        # test escape_newline
+        self.assertEqual(quote_str("foo\nbar", escape_newline=False), '"foo\nbar"')
+        self.assertEqual(quote_str("foo\nbar", escape_newline=True), '"""foo\nbar"""')
+
+        # non-string values
+        n = 42
+        self.assertEqual(quote_str(n), 42)
+        l = ["foo", "bar"]
+        self.assertEqual(quote_str(l), ["foo", "bar"])
+        self.assertEqual(quote_str(('foo', 'bar')), ('foo', 'bar')) 
+	 
+	
+    def test_dump(self):
+        """Test EasyConfig's dump() method."""
+        test_ecs_dir = os.path.join(os.path.dirname(os.path.abspath(__file__)), 'easyconfigs')
+
+        for f in ['toy-0.0.eb', 'goolf-1.4.10.eb', 'ScaLAPACK-2.0.2-gompi-1.4.10-OpenBLAS-0.2.6-LAPACK-3.4.2.eb']:
+            test_ec = os.path.join(self.test_prefix, 'test.eb')
+
+            ec = EasyConfig(os.path.join(test_ecs_dir, f))
+            ec.dump(test_ec)
+            ectxt = read_file(test_ec)
+            name_regex = re.compile('^name = "[^"]*"', re.M)
+            description_regex = re.compile('^description = "[^"]*"', re.M)
```

Comment:
the regex here is kind of flawed... `""` would match too, for example, and since `[^"]*` also matches the empty string, you're not checking the full string anyway...

maybe just go with `'^description = ["\']'` ?